#VRML_SIM R2020a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.414460951885007 -0.8165970359842091 -0.40173548534375014 1.7564517636422214
  position -1.89399045515011 2.6221575921511953 0.03295992668711149
}
TexturedBackground {
  texture "factory"
}
TexturedBackgroundLight {
}
DEF epuck E-puck {
  translation 0.366928 3.99681e-15 0.0121104
  rotation 4.9563696043727e-06 0.9999999201780133 -0.0003995239681092006 1.57078
  controller "lab4_task4_noise"
  supervisor TRUE
  camera_width 50
  turretSlot [
    InertialUnit {
      rotation 0 1 0 3.14
    }
    DEF righthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
          controlPID 8 2 0.5
        }
        PositionSensor {
          name "right wheel sensor"
          noise 0.1
        }
      ]
      endPoint Solid {
        translation 0.04500995778380096 0.025013380758883635 -2.4671929852070174e-05
        rotation 0.9999993203499804 0.0010052734270078172 -0.0005905293508524928 4.6368371220546685
        children [
          DEF WHEEL Transform {
            translation -0.007 0.01 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "right wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        rotationStep 0.25
      }
    }
    DEF lefthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
          controlPID 8 0 0
        }
        PositionSensor {
          name "left wheel sensor"
          noise 0.1
        }
      ]
      endPoint Solid {
        translation -0.04501012942703836 0.024977671977279067 -4.5733976162035025e-05
        rotation -0.9999982113395169 0.00047275268613733136 -0.0018313444964861226 1.2425810202889593
        children [
          DEF WHEEL Transform {
            translation 0.007 0 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.1 0.1 0.1
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "left wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF front_ds DistanceSensor {
      translation 0 0 -0.045
      rotation 0 1 0 1.5708
      name "front_ds"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF right_ds DistanceSensor {
      translation 0.035 -0.01 -0.01
      name "right_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF left_ds DistanceSensor {
      translation -0.035 -0.01 -0.01
      rotation 0 1 0 3.14159
      name "left_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    Camera {
      translation 0 -0.01 -0.045
      name "camera1"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
WoodenBox {
  translation 0.106263 2.22045e-16 -0.0861506
  rotation 0 1 0 -1.5707953071795862
  name "wooden box(5)"
  size 0.16 0.2 0.1
}
WoodenBox {
  translation -0.146461 2.22045e-16 0.191323
  rotation 0 1 0 -1.5707953071795862
  name "wooden box(1)"
  size 0.16 0.2 0.1
}
WoodenBox {
  translation 0.179941 2.22045e-16 0.253117
  rotation 0 1 0 3.14159
  name "wooden box(2)"
  size 0.16 0.2 0.1
}
WoodenBox {
  translation -0.153405 2.22045e-16 -0.249404
  rotation 0 1 0 3.14159
  name "wooden box(3)"
  size 0.16 0.2 0.1
}
DEF RedCylinder Solid {
  translation 0.508 0.2 0.508
  rotation -0.17730642736867652 0.9782184953720645 0.10793982641161912 -9.44154186989369e-16
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.381
        radius 0.08
      }
    }
  ]
  name "solid(1)"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  locked TRUE
  recognitionColors [
    1 1 0
  ]
}
DEF YellowCylinder Solid {
  translation 0.508 0.2 -0.508
  rotation -0.17730642736867652 0.9782184953720645 0.10793982641161912 -9.44154186989369e-16
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.381
        radius 0.08
      }
    }
  ]
  name "solid(4)"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  locked TRUE
  recognitionColors [
    1 1 0
  ]
}
DEF GreenCylinder Solid {
  translation -0.508 0.2 -0.508
  rotation -0.17730642736867652 0.9782184953720645 0.10793982641161912 -9.44154186989369e-16
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.381
        radius 0.08
      }
    }
  ]
  name "solid(3)"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  locked TRUE
  recognitionColors [
    1 1 0
  ]
}
DEF BlueCylinder Solid {
  translation -0.508 0.2 0.508
  rotation -0.17730642736867652 0.9782184953720645 0.10793982641161912 -9.44154186989369e-16
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.381
        radius 0.08
      }
    }
  ]
  name "solid(5)"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  locked TRUE
  recognitionColors [
    1 1 0
  ]
}
DEF area RectangleArena {
  floorSize 1.016 1.016
  floorTileSize 0.508 0.508
  wallAppearance BrushedAluminium {
    colorOverride 0 0 0
  }
}
